receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -145,6 +149,13 @@ func (tr *transaction) Commit() error {
 		dropped := 0
 		metrics, dropped = NewMetricsAdjuster(tr.jobsMap.get(tr.job, tr.instance), tr.logger).AdjustMetrics(metrics)
 		droppedTimeseries += dropped
+	} else if tr.useStartTimeMetric {
+		if tr.metricBuilder.startTime == 0.0 {
+			metrics = []*metricspb.Metric{}
+			droppedTimeseries = numTimeseries
+		} else {
+			adjustStartTime(tr.metricBuilder.startTime, metrics)
+		}
```

Comment:
IMO, it would be more readable to have following
```go
	if tr.useStartTimeMetric {
 		// adjustStartTime
	} else {
		// AdjustMetrics. jobsMap has to be non-nil in this case.
	}
```

Reply:
Done